api/v1beta1/tinkerbellmachinetemplate_webhook.go

```diff
@@ -55,7 +55,12 @@ func (m *TinkerbellMachineTemplate) ValidateUpdate(old runtime.Object) error {
 	oldTinkerbellMachineTemplate, _ := old.(*TinkerbellMachineTemplate)
 
 	if !reflect.DeepEqual(m.Spec, oldTinkerbellMachineTemplate.Spec) {
-		return apierrors.NewBadRequest("TinkerbellMachineTemplate.Spec is immutable")
+		newTemplateSpec := m.Spec.Template.Spec
+		oldTemplateSpec := oldTinkerbellMachineTemplate.Spec.Template.Spec
+
+		if !reflect.DeepEqual(newTemplateSpec.HardwareAffinity, oldTemplateSpec.HardwareAffinity) {
```

Comment:
This change would cause only `HardwareAffinity` to immutable and all the other fields in the [spec](https://github.com/tinkerbell/cluster-api-provider-tinkerbell/blob/main/api/v1beta1/tinkerbellmachine_types.go#L32) are now mutable.

Was that intentional? Because many fields like `ProviderID`, `HardwareName` might benefit from being immutable.

Reply:
I agree, it doesn't seem to have the desired outcome. I think it'd be useful to introduce some tests also.

Reply:
Not sure what you mean by "doesn't seem to have the desired outcome". Would help to be more specific. Either way, this PR would soon be closed. I have found a better way to accomplish this. But in the future, templateOverride sitting in TinkerbellMachineTemplate as a string blob and containing dynamic values needs to be addressed in general.

Reply:
I was just agreeing. @pokearu's comment has the specifics but to be explicit I mean to say the change would result in much more than just templateOverride being immutable.